relay-server/src/actors/store.rs

```diff
@@ -63,30 +63,48 @@ fn make_distinct_id(s: &str) -> Uuid {
 
 impl StoreForwarder {
     pub fn create(config: Arc<Config>) -> Result<Self, ServerError> {
-        let mut client_config = ClientConfig::new();
-        for config_p in config.kafka_config() {
-            client_config.set(config_p.name.as_str(), config_p.value.as_str());
-        }
+        // Temporary map used to deduplicate kafka producers
+        let mut reused_producers: BTreeMap<_, Arc<_>> = BTreeMap::new();
+
+        let producers = config
+            .kafka_topics()
```

Comment:
This elegantly comes with the advantage that you never connect to a broker that's unreferenced from the topic assignments. It's a bit of a stretch that you need to now expose a generic topic map from `relay-config` just for that, though.